core/src/mast/serialization/mod.rs

```diff
@@ -232,35 +229,105 @@ impl Deserializable for MastForest {
             mast_forest
         };
 
+        let basic_block_decorators: Vec<(usize, DecoratorList)> = read_block_decorators(source)?;
+        for (node_id, decorator_list) in basic_block_decorators {
+            let node_id = MastNodeId::try_from((node_id, &mast_forest))?;
```

Comment:
Nit: Instead of the TryFrom with a tuple, I would add a `from_usize_safe` method analogous to `from_u32_safe`. This seems more in line with the existing API and I don't think there's a benefit to implementing the trait.